pkg/web/api/health_checker.go

```diff
@@ -17,28 +17,76 @@ package api
 import (
 	"context"
 
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/health/grpc_health_v1"
+	grpcstatus "google.golang.org/grpc/status"
 )
 
-type HealthChecker struct {
+type Checker interface {
+	Check(ctx context.Context) error
+}
+
+// HealthServer implements the gRPC health service.
+// Using the HealthServer, it's possible to check
+// the gRPC server's overall health, but also the
+// health of the individual gRPC services.
+// For more information, see: https://github.com/grpc/grpc/blob/master/doc/health-checking.md
+type HealthServer struct {
 	grpc_health_v1.UnimplementedHealthServer
+	checkers map[string]Checker
+	log      log.CtxLogger
```

Comment:
Nitpick: throughout the codebase we use the var name `logger` for this type, would be nice to have this consistent with the rest.